packages/typespec-ts/test/unit/parametersGenerator.spec.ts

```diff
@@ -3,92 +3,153 @@ import { emitParameterFromCadl } from "./util/emitUtil.js";
 import { assertEqualContent } from "./util/testUtil.js";
 
 describe("Parameters.ts", () => {
-  describe("handle query apiVersion", () => {
-    it("should't generate apiVersion if there's a client level apiVersion", async () => {
-      const parameters = await emitParameterFromCadl(
-        `
-        model ApiVersionParameter {
-          @query
-          "api-version": string;
-        }
-        op test(...ApiVersionParameter): string;
-        `
-      );
-      assert.ok(parameters);
-      assertEqualContent(
-        parameters?.content!,
-        `
-          import { RequestParameters } from "@azure-rest/core-client";
-          
-          export type TestParameters =  RequestParameters;
+  describe("query parameters", () => {
+    describe("apiVersion in query", () => {
+      it("should't generate apiVersion if there's a client level apiVersion", async () => {
+        const parameters = await emitParameterFromCadl(
           `
-      );
+          model ApiVersionParameter {
+            @query
+            "api-version": string;
+          }
+          op test(...ApiVersionParameter): string;
+          `
+        );
+        assert.ok(parameters);
+        assertEqualContent(
+          parameters?.content!,
+          `
+            import { RequestParameters } from "@azure-rest/core-client";
+            
+            export type TestParameters =  RequestParameters;
+            `
+        );
+      });
+
+      it("should generate apiVersion if there's no client level apiVersion", async () => {
+        const parameters = await emitParameterFromCadl(
+          `
+          model ApiVersionParameter {
+            @query
+            "api-version": string;
+          }
+          op test(...ApiVersionParameter): string;
+          `,
+          false,
+          true
+        );
+        assert.ok(parameters);
+        assertEqualContent(
+          parameters?.content!,
+          `
+            import { RequestParameters } from "@azure-rest/core-client";
+            
+            export interface TestQueryParamProperties {
+              "api-version": string;
+            }
+            
+            export interface TestQueryParam {
+              queryParameters: TestQueryParamProperties;
+            }
+            
+            export type TestParameters = TestQueryParam & RequestParameters;
+            `
+        );
+      });
     });
-
-    it("should generate apiVersion if there's no client level apiVersion", async () => {
-      const parameters = await emitParameterFromCadl(
-        `
-        model ApiVersionParameter {
-          @query
-          "api-version": string;
-        }
-        op test(...ApiVersionParameter): string;
-        `,
-        false,
-        true
-      );
-      assert.ok(parameters);
-      assertEqualContent(
-        parameters?.content!,
-        `
+    describe("other parameters in query", () => {
+      it("should generate user-custom-query ", async () => {
```

Comment:
Not sure what's the point of adding this test ?

Reply:
this is existing test case I just re-org this one.